Documenting Applications Mapping file sample configurations (#35)
IBM/dbb-git-migration-modeler: docs/AdvancedInformation.md

  * For each artifact of the application, an entry is added in [languageConfigurationMapping.properties](https://github.com/IBM/dbb-zappbuild/blob/main/samples/application-conf/languageConfigurationMapping.properties), which maps the artifact with its Language Configuration defined in [build-conf/language-conf](https://github.com/IBM/dbb-zappbuild/tree/main/build-conf/language-conf)

> # Setup the DBB Git Migration Modeler utility

Reviewer: ```suggestion
# Setting up the DBB Git Migration Modeler utility
```

| PIPELINE_CI | Pipeline technology used - either `AzureDevOps`, `GitlabCI`, `Jenkins` or `GitHubActions` | `AzureDevOps` |

> # Configure Migration Modeler input files

Reviewer: ```suggestion
# Configuring the Migration Modeler input files 
```

The utility is operating on a set of provided PDS libraries that contain a copy of the codebase to be migrated. Depending on your configuration, it may be required to unload the source files from the legacy SCM's storage, prior to using the DBB Git Migration Modeler. These datasets should be extracted from the legacy SCM system, using a SCM-provided utility or mechanism. The list of datasets that contain source files is defined [Applications Mapping file](../samples/applicationsMapping.yaml) for a set of applications.

Also, the latest steps of the whole migration process are performing a preview build and the packaging of existing artifacts. These existing artifacts (loadmodules, DBRMs, and any other artifacts meant to be deployed belonging to the applications) are expected to be found in datasets, following the naming convention in dbb-zAppBuild for output datasets. Typically, loadmodules are stored in to a `HLQ.LOAD` library, object decks in a `HLQ.OBJ` library and DBRMS in a `HLQ.DBRM` library. The HLQ used during this phase is provided through the `APPLICATION_ARTIFACTS_HLQ` environment variable defined during the execution of the [Setup script](./Setup.sh).

> ## Define Applications Mapping files

Reviewer: The main README file needs to be changed as well as it contains a reference to this chapter as a link I think.

      - ........
~~~~

It is common to have shared files that do not belong to an application and are not assigned to applications - for instance copybooks or include files.

> To pass information to Migration Modeler about datasets containing the shared copybooks or include files in this scenario, create an Applications Mapping file (* UNASSIGNED.yaml*) pointing to the shared libraries, but don't list any applications:

Reviewer: ```suggestion
To pass information to the Migration Modeler about datasets containing the shared copybooks or include files in this scenario, create an Applications Mapping file (*SHARED.yaml*) pointing to the shared libraries, but don't list any applications:
```

~~~~YAML
datasets:
  - APPS.COPYLIB
  - APPS.INCLIB
applications: []
~~~~

> When running the Migration Modeler with these Applications Mapping files, all the artifacts found in the input datasets (CATMAN.COBOL, CATMAN.COPY and CATMAN.BMS) will be assigned to the Catalog Manager application, and artifacts found in APPs.COPYLIB and APPS.INCLIB will be assigned to the special application called *UNASSIGNED*.

Reviewer: Have you tested that, all the artifacts from the UNASSIGNED application and the artifacts from other applications for no naming convention were found, they all end up in the UNASSIGNED application correctly?

Author: Yes, I did

      - ........
~~~~

If shared code following naming conventions and is planned to managed in their own git repository, you can also provide naming conventions.

### Combining use cases

> You can combine the above scenarios to configure input to Migration Modeler. For instance, a given library contains artifacts from one application, while other libraries contain files from multiple applications. Or you need to apply specific naming conventions for include files.

Reviewer: ```suggestion
You can combine the above scenarios to configure Applications mapping files to use with the Migration Modeler. For instance, a given library contains artifacts from one application, while other libraries contain files from multiple applications. Or you need to apply specific naming conventions for include files.
```

> You can combine the above scenarios to configure Applications mapping files to use with the Migration Modeler. For instance, a given library contains artifacts from one application, while other libraries contain files from multiple applications. Or you need to apply specific naming conventions for include files.

Reviewer: ```suggestion
You can combine the above scenarios to configure Applications Mapping files to use with the Migration Modeler. For instance, a given library contains artifacts from one application, while other libraries contain files from multiple applications. Or you need to apply specific naming conventions for include files.
```

In that case, the solution is to configure multiple Applications Mapping files:
- One Applications Mapping file would contain the definition for the datasets having artifacts belonging to only one application
- A second Applications Mapping file would contain the definitions for the datasets having artifacts from multiple applications.

> The [Migration-Modeler-Start script](./src/scripts/Migration-Modeler-Start.sh) will process all available Applications Mapping files. 

Reviewer: ```suggestion
The [Migration-Modeler-Start script](../src/scripts/Migration-Modeler-Start.sh) will process all available Applications Mapping files. 
```
